pkg/expr/sql_command_test.go

```diff
@@ -25,3 +31,115 @@ func TestNewCommand(t *testing.T) {
 		return
 	}
 }
+
+func TestExecuteWithinLimit(t *testing.T) {
+	cmd, err := NewSQLCommand("a", "select a from foo, bar")
+	if err != nil {
+		t.Fatalf("Failed to create SQL command: %v", err)
+	}
+	cmd.limit = 2
+
+	frame := data.NewFrame("a", data.NewField("a", nil, []string{"1", "2"}))
+	vars := mathexp.Vars{}
+	vars["foo"] = mathexp.Results{
+		Values: mathexp.Values{mathexp.TableData{Frame: frame}},
+	}
+
+	_, err = cmd.Execute(context.Background(), time.Now(), vars, &testTracer{})
+	if err != nil {
+		t.Fatalf("Expected no error, got: %v", err)
+	}
+}
+
+func TestExecuteWithinLimitWithMultipleFrames(t *testing.T) {
+	cmd, err := NewSQLCommand("a", "select a from foo, bar")
+	if err != nil {
+		t.Fatalf("Failed to create SQL command: %v", err)
+	}
+	cmd.limit = 4
+
+	frame1 := data.NewFrame("a", data.NewField("a", nil, []string{"1", "2"}))
+	frame2 := data.NewFrame("b", data.NewField("a", nil, []string{"3", "4"}))
+	vars := mathexp.Vars{}
+	vars["foo"] = mathexp.Results{
+		Values: mathexp.Values{mathexp.TableData{Frame: frame1}},
+	}
+	vars["bar"] = mathexp.Results{
+		Values: mathexp.Values{mathexp.TableData{Frame: frame2}},
+	}
+
+	_, err = cmd.Execute(context.Background(), time.Now(), vars, &testTracer{})
+	if err != nil {
+		t.Fatalf("Expected no error, got: %v", err)
+	}
+}
+
+func TestExecuteExceedsLimit(t *testing.T) {
```

Comment:
You could probably pick this test or the one below, I don't think you need both :thinking:

Reply:
Thanks. Single frame cases must always be a subset of multi-frame cases I think - so maybe I'll remove both of the single-frame cases. WDYT?